src/AutoRest.CSharp/Common/Output/Models/Types/ObjectTypeProperty.cs

```diff
@@ -164,6 +167,126 @@ public static FormattableString CreateDefaultPropertyDescription(string nameToUs
 
         public SourcePropertySerializationMapping? SerializationMapping { get; }
 
+        /// <summary>
+        /// This method attempts to retrieve the description for each of the union type items. For items that are lists,
+        /// the description will include details about the element type.
+        /// For items that are literals, the description will include the literal value.
+        /// </summary>
+        /// <param name="unionItems">the list of union type items.</param>
+        /// <returns>A list of FormattableString representing the description of each union item.
+        /// </returns>
+        public static IReadOnlyList<FormattableString> GetUnionTypesDescriptions(IList<CSharpType> unionItems)
+        {
+            var values = new List<FormattableString>();
+
+            foreach (CSharpType item in unionItems)
+            {
+                if (item != null)
+                {
+                    FormattableString description = $"<description><see cref=\"{item}\"/></description>";
+
+                    if (item.IsLiteral && item.Literal?.Value != null)
+                    {
+                        var literalValue = item.Literal.Value.Value;
+                        if (item.FrameworkType == typeof(string))
+                        {
+                            description = $"<description>{literalValue:L}</description>";
+                        }
+                        else
+                        {
+                            description = $"<description>{literalValue}</description>";
+                        }
```

Comment:
why we have to have this if? The writer could handle it properly if the value is string or not string therefore we could always have `{literalValue:L}`